src/components/questions/RecordUpload.tsx

```diff
@@ -0,0 +1,158 @@
+import AudioRecorderPlayer, {
+  AVEncoderAudioQualityIOSType,
+  AVEncodingOption,
+  AudioEncoderAndroidType,
+  AudioSet,
+  AudioSourceAndroidType,
+  RecordBackType,
+  PlayBackType,
+} from 'react-native-audio-recorder-player';
+import { Alert, PermissionsAndroid, Platform, StyleSheet, View } from 'react-native';
+import React, { Component } from 'react';
+
+import Icon from 'react-native-vector-icons/FontAwesome';
+import COLORS from '../../assets/colors';
+
+interface State {
+  isPlaying: boolean;
+  isRecording: boolean;
+  onVoiceSelected: string;
+}
+class RecordUpload extends Component<any, State> {
+  private audioRecorderPlayer: AudioRecorderPlayer;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
```

Comment:
Is dit hetzelfde als een React.useState?

Reply:
ja precies hetzelfde maar inplaats van in een function component word dit zo in een class gedaan.